common/ast/ast.go

```diff
@@ -533,3 +533,118 @@ func (hv heightVisitor) maxEntryHeight(entries ...EntryExpr) int {
 	}
 	return max
 }
+
+
+// Children returns a map from expression id to a list of the ids of its direct children.
+func Children(a *AST) map[int64][]int64 {
```

Comment:
normalized parent/child traversal is provided by the navigable AST types.

`
navExpr := ast.NavigateExpr(cAST, expr)
`

Could you describe what you need here that can't be accomplished with NavigableExpr? If there is a tradeoff to using id maps over the Navigable* types, can you document that here?

common/ast/navigable.go